gfwanalysis/services/geostore_service.py

```diff
@@ -10,97 +10,71 @@ class GeostoreService(object):
 
     @staticmethod
-    def create(geojson):
-        config = {
-            'uri': '/geostore',
-            'method': 'POST',
-            'body': {
-                'geojson': geojson
-            }
-        }
-        return GeostoreService.execute(config)
+    def create(geojson, api_key):
+        return GeostoreService.execute(
+            "/v2/geostore", api_key, body=geojson, method="POST"
+        )
 
     @staticmethod
-    def get(geostore):
-        config = {
-            'uri': '/geostore/' + geostore,
-            'method': 'GET'
-        }
-        return GeostoreService.execute(config)
+    def get(geostore, api_key):
+        uri = f"/v2/geostore/{geostore}"
+        return GeostoreService.execute(uri, api_key)
 
     @staticmethod
-    def get_national(iso):
+    def get_national(iso, api_key):
         # Lookup table here to reduce complexity
         # logging.info('[geostore service]: in get_national function')
         simplification = admin_0_simplify(iso)
         # logging.info(f'[geostore service]: simplification={simplification}')
         if simplification:
-            url = f'/v2/geostore/admin/{iso}?simplify={simplification}'
+            url = f"/v2/geostore/admin/{iso}?simplify={simplification}"
         else:
-            url = f'/v2/geostore/admin/{iso}'
-        config = {
-            'uri': url,
-            'method': 'GET',
-            'ignore_version': True
-        }
-        return GeostoreService.execute(config)
+            url = f"/v2/geostore/admin/{iso}"
+        return GeostoreService.execute(url, api_key)
 
     @staticmethod
-    def get_subnational(iso, id1):
+    def get_subnational(iso, id1, api_key):
         # logging.info('[geostore service]: in get_subnational function')
         simplification = admin_1_simplify(iso, id1)
         # logging.info(f'[geostore service]: simplification={simplification}')
         if simplification:
-            url = f'/v2/geostore/admin/{iso}/{id1}?simplify={simplification}'
+            url = f"/v2/geostore/admin/{iso}/{id1}?simplify={simplification}"
         else:
-            url = f'/v2/geostore/admin/{iso}/{id1}'
-        config = {
-            'uri': url,
-            'method': 'GET',
-            'ignore_version': True
-        }
-        return GeostoreService.execute(config)
+            url = f"/v2/geostore/admin/{iso}/{id1}"
+        return GeostoreService.execute(url, api_key)
 
     @staticmethod
-    def get_regional(iso, id1, id2):
-        config = {
-            'uri': '/v2/geostore/admin/' + iso + '/' + id1 + '/' + id2,
-            'method': 'GET',
-            'ignore_version': True
-        }
-        return GeostoreService.execute(config)
+    def get_regional(iso, id1, id2, api_key):
+        uri = f"/v2/geostore/admin/{iso}/{id1}/{id2}"
+        return GeostoreService.execute(uri, api_key)
 
     @staticmethod
-    def get_use(name, use_id):
-        config = {
-            'uri': '/geostore/use/' + name + '/' + use_id,
```

Comment:
by default, I think non-versioned URLs are routed to v1 - thus this adds a potential BC break. While I did notice that some calls to geostore use v2 explicitly, I don't think that warrants a potential BC break. Do you have a rationale for this change other than that?

Reply:
Grappled with this choice but saw user geostores of both versions are stored in the same table and `GET` responses are identical. But will defer to you and update.

Reply:
> based on the geostore [doc](https://resource-watch.github.io/doc-api/reference.html#overview-of-available-endpoints) that v2 should be used except in few situations

I couldn't find this explicit recommendation on the docs, but it's probably there in some form, and it's typically aimed at new implementations. It's not meant to be read as "v2 is a drop-in replacement for v1". It's meant to read as "if you are using geostore for the first time, and don't know which version to use, use v2". If there is any text in the docs hinting at v1 and v2 being drop-in replacements, it should be reviewed - I suspect it may be false (but would have to confirm).

The docs do explicitly mention that different versions of geostore endpoints use different GADM versions. I don't recall exactly how these may or may not affect the results of different endpoints, but the bottom line is that there _may_ be a difference that propagates here. So the question is not so much about whether or not there is a difference between geostore v1 and v2 - until proven otherwise, assume there is - but rather if you have tested that your code changes here account for those changes.

Reply:
There is nothing preventing end users of this MS from requesting/expecting v2 version of geostore and the MS is silently going to return analysis based on v1. So not sure whether it's worse to break backward or forward compatibility, went w/ my interpretation of the doc in the choice of v2. I have reverted this to stay within scope and I don't have the bandwidth to test the impact, if any, of each choice. This may not even matter if we find this MS is obsolete after we put in the API keys;)